### PR TITLE
Feature: Add donate button caption to form settings

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
## Description

This PR adds a new form setting, allowing users to edit the button caption for the 'donate now' button/submit button. The applied changes will display on all 3 templates.

## Visuals

Screen Shot 2023-08-17 at 8 58 26 AM

https://github.com/impress-org/givewp/assets/75056371/f07ca011-0868-4d31-9e17-98a2d3039962

## Testing Instructions

- Add a v3 form.
- Use the Form Settings tab and use the Donate Button setting to update the button caption.
- Verify the button caption has been updated for all 3 form templates.

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205328019764976

### CODE REVIEW

File: src/DonationForms/resources/app/form/MultiStepForm/components/SubmitButton.tsx [hunk 1]
```diff
@@ -3,14 +3,12 @@ import {__} from '@wordpress/i18n';
 /**
  * @since 3.0.0
  */
-export default function SubmitButton({
-    isSubmitting,
-    submittingText = __('Submitting…', 'give'),
-    buttonText = __('Donate Now', 'give'),
-}) {
+export default function SubmitButton({isSubmitting, submittingText = __('Submitting…', 'give')}) {
+    const {donateButtonCaption} = window.givewp.form.hooks.useDonationFormSettings();
```
Comment (jonwaldstein) [inline src/DonationForms/resources/app/form/MultiStepForm/components/SubmitButton.tsx:7]: @JoshuaHungDinh i'm getting a type error on this, please add to `PropTypes` in `src/DonationForms/resources/app/store/form-settings/index.tsx`

```
donateButtonCaption: string;
```

**Review by jonwaldstein — CHANGES_REQUESTED** [review]

File: src/FormBuilder/resources/js/form-builder/src/components/sidebar/Primary/index.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import TabPanel from '../TabPanel';
 
 import {
     CustomStyleSettings,
+    DonateButton,
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/FormBuilder/resources/js/form-builder/src/components/sidebar/Primary/index.tsx]: ah, late change I pushed last night. Thank you! resolved: https://github.com/impress-org/givewp/pull/6883/commits

File: src/DonationForms/resources/app/form/MultiStepForm/components/SubmitButton.tsx (hunk 1 above)
Comment (JoshuaHungDinh) [reply src/DonationForms/resources/app/form/MultiStepForm/components/SubmitButton.tsx:7]: Resolved: 0eb16bef6c58361a0ddd1b855bd8398c8b0499eb

**Review by jonwaldstein — APPROVED** [review]

**Review by rickalday — APPROVED** [review]
Passed manual QA tests.

**Pull request merged**